pgdog/src/frontend/router/parser/query.rs

```diff
@@ -226,6 +244,13 @@ impl QueryParser {
         let mut command = match root.node {
             // SELECT statements.
             Some(NodeEnum::SelectStmt(ref stmt)) => {
+                // Check for side-effect functions in the target list
+                if Self::has_side_effect_function(stmt) {
+                    debug!("Query contains side-effect function, routing to primary.");
+                    return Ok(Command::Query(Route::write(None)));
```

Comment:
This makes it a cross-shard query. I don't think we want that, necessarily, especially for advisory locks. You'd probably want to pick a shard that handles it, since advisory locks on different shards won't actually mutually exclude queries.